config/read.go

```diff
@@ -771,32 +781,26 @@ func Read(logger *util.Logger, filename string) (Config, error) {
 				SystemScope: config.SystemScope,
 			}
 
-			if config.GetDbName() != "" {
-				// Ensure we have no duplicate identifiers within one collector
-				skip := false
-				for _, server := range conf.Servers {
-					if config.Identifier == server.Identifier {
-						skip = true
-					}
-				}
-				if skip {
+			// Ensure we have no duplicate identifiers within one collector
+			for _, server := range conf.Servers {
+				if config.Identifier == server.Identifier {
 					logger.PrintError("Skipping config section %s, detected as duplicate. Note: To monitor multiple databases on the same server, db_name accepts a comma-separated list.", config.SectionName)
-				} else {
-					conf.Servers = append(conf.Servers, *config)
+					continue
 				}
 			}
 
 			if config.DbURL != "" {
 				_, err := url.Parse(config.DbURL)
 				if err != nil {
-					prefixedLogger := logger.WithPrefix(config.SectionName)
-					prefixedLogger.PrintError("Could not parse db_url; check URL format and note that any special characters must be percent-encoded")
+					return conf, fmt.Errorf("Could not parse db_url in section %s; check URL format and note that any special characters must be percent-encoded", config.SectionName)
```

Comment:
I think in practice this would never be hit, since `config.GetDbName()` ends up being empty in such cases ( (https://github.com/pganalyze/collector/blob/main/config/config.go#L459), leading to an early return a few lines up.

Reply:
Oh good catch. I can put it before the `GetDbName` check, right? I think it's still a useful error to have, since we can give more concrete guidance in this case.

Reply:
And I guess similar to the other comment, I'll make this a `PrintError`.

Reply:
Yeah, I think you should be able to do that check early, and agreed, its useful to have.